docs/madr/decisions/073-spiffe-compliance.md

```diff
@@ -0,0 +1,136 @@
+
+### As a user, it should not become harder to enable or manage mTLS in either single-zone or multi-zone deployments than it is today.
+
+Currently, users can enable mTLS in both single-zone and multi-zone deployments just by configuring it in the Mesh resource. As we introduce features like trust domains, external identities, or federation, we must preserve this simplicity and avoid adding unnecessary complexity to the mTLS setup process.
+
+### As a user, I want to be able to communicate with applications in the different trust domain.
+
+In the presence of multiple trust domains. Users should be able to trust each domains.
+
+### As a Mesh operator, I want to be able to define trustDomain at different levels.
+
+It makes sense to allow users to define a trust domain for a specific part of the infrastructure: whether that's a zone or an entire mesh.
+By assigning unique trust domains per cluster, we achieve security isolation: if one cluster is compromised, the others remain secure. Additionally, it helps with identity scoping, since we can clearly determine the origin of each identity. Another important aspect is enabling rotation or migration. During the migration, we could use two different trust domains to facilitate the transition from one identity provider to another.
+
+Default trust domain: `spiffe://<mesh>.<clusterId>.kuma.io/`
+
+### As a Mesh operator, I want to be able to migrate some workloads to another trust domain without interrupting traffic.
+
+I have a workloads in a trust domain `example.com` and I want to move them to `example.org`. I should be able to do it and don't break the traffic.
+
+### As a user, I want certificate to be valid for a short period of time
+
+Based on the [specification](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-workload-api) certificates should be short-lived.
+
+### As a user, I want to migrate from Kuma non-SPIFFE complient certificates to SPIRE
+
+We should allow users to migrate from Kuma issued certificates, which are not SPIFFE-compliant, to using SPIRE. This may require an intermediate step to first migrate to SPIFFE-compliant certificates, but the details will be specified later.
+
+## Future User Stories (things that are going to be implemented in future releases)
+
+### As a user, I want to have a SPIFFE-compliant SAN in a non-Kubernetes environment
+
+It would be beneficial to support connectivity between different environments. Unfortunately, our services currently lack proper identity on Universal, which makes this challenging.
+
+### As a user, I want to federate different trust domains using the Federation API exposed by the control plane
+
+It should be possible to federate different trust domains without creating additional resources, only by using the federation endpoint exposed by the control plane. This would be a valuable feature, enabling interoperability between different SPIFFE-compliant systems.
```

Comment:
The federation endpoint is part of SPIRE not spiffe no?

Reply:
https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE_Federation.md#2-introduction